### PR TITLE
Enable changing the unit system

### PR DESCRIPTION
**Pull request by greschd**
Allow setting the `unit_system` attribute on the model.

The `UnitSystemType` enum is extended by a value `FROM_FILE = "from_file"`,
which is an alias to `UNDEFINED`. This allows for using the more descriptive
`"from_file"` when loading an FE model.
When getting the `unit_system` attribute from the model, the `FROM_FILE`
is never returned; the "primary" value `UNDEFINED` is used instead, as before.

### CODE REVIEW

File: tests/unittests/test_model.py [hunk 1]
```diff
@@ -265,12 +267,45 @@ def test_modeling_ply_export(acp_instance, minimal_complete_model, xfail_before)
...
+        "si": 1e-3,
+        "bin": 0.03937008,
+        "bft": 0.00328084,
+        "umks": 1e3,
+    }
+
+    with raises_before_version("25.1"):
+        if unit_system in (UnitSystemType.UNDEFINED, UnitSystemType.FROM_FILE):
+            with pytest.raises(ValueError):
+                minimal_complete_model.unit_system = unit_system
```
Comment (greschd) [inline tests/unittests/test_model.py:302]: Will fail until the corresponding backend PR is merged.

**Comment by greschd** [comment]
@roosre this PR is ready for review, but needs the backend unit system changes for CI to pass

File: src/ansys/acp/core/_tree_objects/model.py [hunk 2]
```diff
@@ -269,8 +272,20 @@ def _create_stub(self) -> model_pb2_grpc.ObjectServiceStub:
     minimum_analysis_ply_thickness: ReadWriteProperty[float, float] = grpc_data_property(
         "properties.minimum_analysis_ply_thickness"
     )
-    unit_system = grpc_data_property_read_only(
-        "properties.unit_system", from_protobuf=unit_system_type_from_pb
+
+    @staticmethod
+    def _set_unit_system_data_attribute(pb_obj: ObjectInfo, name: str, value: _PROTOBUF_T) -> None:
```
Comment (roosre) [inline src/ansys/acp/core/_tree_objects/model.py:277]: Shouldn't we handle that in the backend?

**Review by roosre — APPROVED** [review]

File: src/ansys/acp/core/_tree_objects/model.py (hunk 2 above)
Comment (greschd) [reply src/ansys/acp/core/_tree_objects/model.py:277]: It is handled in the backend, in the sense that if the `PUT` request contains _only_ the unit system it will transform the existing `minimum_analysis_ply_thickness`. 

However, if we don't remove the field here, it will transform as (for example):

- start with units: MPA, minimum_analysis_ply_thickness=1e-6 (mm)
- PUT: units: MKS, minimum_analysis_ply_thickness=1e-6 (m)

Since the PUT request can change both the unit system and minimum_analysis_ply_thickness simultaneously, it is interpreted as being internally consistent (i.e., in the new units). 

So, in short: to let the backend do the conversion, we need to ensure we don't send it along in the PUT request.
